### PR TITLE
Registry System.

### PR DESCRIPTION
**Pull request by MarcasRealAccount**
### Registry System
You can now register elements for later use.
Using `Registry<T>::RegisterElement()` or `Registry<T>::GetElement()`, where `T` is the type you want.
Though you still have to register it manually in a cpp file by doing `Registry<T>* Registry<T>::m_Instance = new Registry<T>();`.

### CODE REVIEW

**Comment by Gaztin** [comment]
Can you give one or a few examples of where this would be used?

**Comment by MarcasRealAccount** [comment]
The registry system would be used to store the base Items and Blocks such that it's easy to handle storing an indefinite number of Items or Blocks, though It could be used for registering other data as well.
It does not have to be used only for Items and Blocks.

If you've ever looked at how Minecraft handles registering blocks and items, it does something similar but to a more advanced degree that what I created here, but it should function the same nonetheless.

So basically we would do something like this:
```cpp
// Registry.cpp
Registry<Item>* Registry<Item>::m_Instance = new Registry<Item>();

// Some file where we need the item registry
Registry::RegisterElement(someItem);
Item* item = Registry<Item>::GetElement("some_item");
```

`Registry<Item>::RegisterElement(someItem);` can be simplified to `Registry::RegisterElement(someItem);` since someItem is of type Item.

**Comment by Gaztin** [comment]
Ah, that's really smart! I like it. Especially since you do explicit instantiation of the registry template, so trying to access an invalid registry will yield a compile-time error!
